### ISSUE TITLE
google_container_cluster.control_plane_endpoints_config - support setting `IPEndpointsConfig.enabled`

### ISSUE THREAD

**Issue by mans0954** [hashicorp/terraform-provider-google#20369]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to a user, that user is claiming responsibility for the issue.
* Customers working with a Google Technical Account Manager or Customer Engineer can ask them to [reach out internally](https://github.com/hashicorp/terraform-provider-google/wiki/Customer-Contact#raising-gcp-internal-issues-with-the-provider-development-team) to expedite investigation and resolution of this issue.

### Description

The GKE API `Cluster.ControlPlaneEndpointsConfig.IPEndpointsConfig` structure has an `enabled` boolean for enabling or disabling direct IP access: https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1/projects.locations.clusters#Cluster.IPEndpointsConfig

Currently this is hard coded to `enabled` in the provider source code: https://github.com/hashicorp/terraform-provider-google/blob/27c9571b60015846bfae7f866c98a50cc1c5042c/google/services/container/resource_container_cluster.go#L4998

With the new [new DNS based endpoint approach for accessing the control plane](https://cloud.google.com/blog/products/containers-kubernetes/new-dns-based-endpoint-for-the-gke-control-plane) there is likely to be interest in being able to disable direct IP access through Terraform.

### New or Affected Resource(s)

* google_container_cluster

### Potential Terraform Configuration

```tf 
  control_plane_endpoints_config {
    dns_endpoint_config {
      allow_external_traffic = true
    }
    ip_endpoint_config {
      enabled = false
   }
  }
```

### References

https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1/projects.locations.clusters#Cluster.IPEndpointsConfig

b/379690021

**Comment by NeckBeardPrince** [comment]
Any chance of having this reviewed?

**Comment by remi-gelinas** [comment]
Agreed, I would like to use the DNS endpoint entirely from private clusters. We need this ability.
